packages/apputils/src/printing.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Any object is "printable" if it implements the `IPrintable` interface.
+ *
+ * To do this it, it must have a method called `Printing.symbol` which returns either a function
+ * to print the object or null if it cannot be printed.
+ *
+ * One way of printing is to use the `printWidget` function, which creates a hidden iframe
+ * and copies the DOM nodes from your widget to that iframe and printing just that iframe.
+ *
+ * Another way to print is to use the `printURL` function, which takes a URL and prints that page.
+ *
+ * See https://github.com/joseluisq/printd#print
+ */
+
+import { Printd, PrintdCallback } from 'printd';
+import { Widget } from '@phosphor/widgets';
+
+export namespace Printing {
+  /**
+   * Function that takes no arguments and when invoked prints out some object or null if printing is not defined.
+   */
+  export type OptionalAsyncThunk = () => Promise<void> | null;
```

Comment:
Can you check if this interface runs into any difficulties in light of the stricter tslint settings from #6125?

Reply:
Sure!

Reply:
I handled a case here and we are good now.